py/selenium/webdriver/remote/webdriver.py

```diff
@@ -82,11 +87,11 @@ def __init__(self, command_executor='http://127.0.0.1:4444/wd/hub',
         if type(self.command_executor) is bytes or isinstance(self.command_executor, str):
             self.command_executor = RemoteConnection(command_executor, keep_alive=keep_alive)
         self._is_remote = True
-        self.session_id = None
-        self.capabilities = {}
+        self.session_id = session_id
+        self.capabilities = capabilities
         self.error_handler = ErrorHandler()
         self.start_client()
-        self.start_session(desired_capabilities, browser_profile)
+        self.start_session(desired_capabilities, browser_profile, force_new=False)
```

Comment:
@lukeis My very own personal opinion, and this is a digression from the actual topic at hand, is that one should strive hard to avoid computational side-effects in class constructors.

That session = FirefoxDriver(…) ends up starting a Firefox process and a new session is an example of this. The benefit of constructors free from side-effects is that instances may be constructed without repercussion and instrumented before they are used, for example to modify the session_id property. I fully understand that you may not want to make these backwards-breaking changes to the API, however, but I thought I should say something about the current design.

To the question at hand: should it be possible to use this client to talk to an existing WebDriver session? You are right in arguing it is a niche use case and I don’t think the complexity this patch offers is worth inflicting upon the already heavily burdened constructor.

On the other hand, I do think it is possible to achieve the same effect the patch author desires, had the original design been so that session_id could be modified before a new session was requested.

I hope this helps!